### PR TITLE
repro: -P option added

### LINKED ISSUE #465: flags in usage not documented in synopsis
In ``repro`` ``-P`` or ``--all-pipelines`` is described in ``usage`` but not documented in ``Synopsis`` . The same case with ``--outs-persist`` and ``--outs-persist-no-cache`` in ``run``.

### PR DESCRIPTION
**Pull request by ryokugyu**
Fix #465

### CODE REVIEW

File: static/docs/commands-reference/run.md [hunk 1]
```diff
@@ -146,6 +146,12 @@ pipeline.
 
   > The `dvc status` command will mention that the file is `not in cache`.
 
+- `-outs-persist`, `--OUTS_PERSIST` - the same as `-o` except that outputs will
```
Comment (efiop) [inline static/docs/commands-reference/run.md]:
```suggestion
- `--outs-persist` - the same as `-o` except that outputs will
```

File: static/docs/commands-reference/run.md [hunk 2]
```diff
@@ -146,6 +146,12 @@ pipeline.
...
+  persist after multiple execution of `dvc repro` command.
+
+- `-outs-persist-no-cache`, `--OUTS_PERSIST_NO_CACHE` - the same as `-O` except
```
Comment (efiop) [inline static/docs/commands-reference/run.md]:
```suggestion
- `--outs-persist-no-cache` - the same as `-O` except
```

**Comment by ryokugyu** [comment]
thanks @efiop 😄

File: static/docs/commands-reference/run.md [hunk 3]
```diff
@@ -146,6 +146,12 @@ pipeline.
 
   > The `dvc status` command will mention that the file is `not in cache`.
 
+- `--outs-persist` - the same as `-o` except that outputs will
```
Comment (efiop) [inline static/docs/commands-reference/run.md]: Maybe just 'the same as `-o` except that outputs will not be removed upon repro'? I'm not so sre about the "after multiple execution" part, it seems confusing.

File: static/docs/commands-reference/repro.md [hunk 4]
```diff
@@ -80,6 +80,9 @@ specified), and updates stage files with the new checksum information.
   file(s) belong(s) to. Use `dvc pipeline show <target>.dvc` to show the parent
   pipeline of a target stage.
 
+- `-P`, `--all-pipelines` - reproduce the entire pipeline(s) for all the stage
```
Comment (shcheklein) [inline static/docs/commands-reference/repro.md]: the entire -> all

File: static/docs/commands-reference/run.md [hunk 5]
```diff
@@ -146,6 +146,12 @@ pipeline.
 
   > The `dvc status` command will mention that the file is `not in cache`.
 
+- `--outs-persist` - the same as `-o` except that outputs will not be removed
```
Comment (shcheklein) [inline static/docs/commands-reference/run.md]: please, remove this - there is PR for this - https://github.com/iterative/dvc.org/pull/217/files

it's not being merged for a reason. These options are still semi-official.

Comment (shcheklein) [reply static/docs/commands-reference/run.md]: btw, that PR is a good example how the option should be described.

Comment (ryokugyu) [reply static/docs/commands-reference/run.md]: @shcheklein remove both `--outs-persist` and `--outs-persist-no-cache` option?

Comment (ryokugyu) [reply static/docs/commands-reference/run.md]: > btw, that PR is a good example how the option should be described.

yes, examples are mentioned also. will try to introduce new information in such a way from next time.

File: static/docs/commands-reference/repro.md [hunk 6]
```diff
@@ -80,6 +80,9 @@ specified), and updates stage files with the new checksum information.
   file(s) belong(s) to. Use `dvc pipeline show <target>.dvc` to show the parent
   pipeline of a target stage.
 
+- `-P`, `--all-pipelines` - reproduce all pipeline(s) for all the stage file(s)
```
Comment (shcheklein) [inline static/docs/commands-reference/repro.md]: why (s), btw? in this case it's just all pipelines for all stage files

File: static/docs/commands-reference/repro.md [hunk 7]
```diff
@@ -80,6 +80,9 @@ specified), and updates stage files with the new checksum information.
   file(s) belong(s) to. Use `dvc pipeline show <target>.dvc` to show the parent
   pipeline of a target stage.
 
+- `-P`, `--all-pipelines` - reproduce pipeline for all the stage file(s) present
```
Comment (shcheklein) [inline static/docs/commands-reference/repro.md]: @ryokugyu it should be as simple as "reproduce all pipelines, for all stage files"

**Pull request merged**